src/InstrumentationEngine.Lib/EnumeratorImpl.h

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "DataContainer.h"
+
+using namespace ATL;
+
+namespace MicrosoftInstrumentationEngine
+{
+    template<class ENUM_INTERFACE, class ENUM_ITEM_TYPE>
+    class CEnumerator : public ENUM_INTERFACE, public CDataContainer
+    {
+    private:
+        // CONSIDER: rewrite this using std::vector instead of array.
+        // I pulled this from my old concord implementation, but since consumers are most likely using std::vector,
+        // it seems like that would be a better impl.
+        DWORD m_dwLength;
+        CAutoVectorPtr<CComPtr<ENUM_ITEM_TYPE>> m_members;
+        DWORD m_dwCurrentIndex;
+
+    public:
+
+        CEnumerator() : m_dwLength(0), m_dwCurrentIndex(0)
+        {
+
+        }
+
+    public:
+        DEFINE_DELEGATED_REFCOUNT_ADDREF(CEnumerator);
+        DEFINE_DELEGATED_REFCOUNT_RELEASE(CEnumerator);
+        STDMETHOD(QueryInterface)(_In_ REFIID riid, _Out_ void **ppvObject) override
+        {
+            return ImplQueryInterface(
+                static_cast<ENUM_INTERFACE*>(this),
+                riid,
+                ppvObject
+                );
+        }
+
+        HRESULT Initialize(
+            _In_ DWORD dwLength,
+            _In_reads_(dwLength) ENUM_ITEM_TYPE** members
+            )
+        {
+            HRESULT hr = S_OK;
+
+            if (dwLength != 0)
+            {
+                m_members.Attach(new CComPtr<ENUM_ITEM_TYPE>[dwLength]);
+                if (!m_members)
+                {
+                    return E_OUTOFMEMORY;
+                }
+
+                for (DWORD i = 0; i < m_dwLength; i++)
```

Comment:
>m_dwLength [](start = 38, length = 10)

m_dewLength = dwLength?

Reply:
I didn't change this code, but just moved the file (actually, I copied the file and forgot to delete the original). I'm going to leave this as-is an log an issue for it.